module/main.c

```diff
@@ -893,94 +872,120 @@ static void handler_HidTimer(s32 data) {
+                                                (tele_get_script_l(edit) < 4)) {
+                                                tele_set_script_l(
+                                                    edit,
+                                                    tele_get_script_l(edit) +
+                                                        1);
+                                            }
                                             if (edit_line <
-                                                SCRIPT_MAX_COMMANDS_) {
+                                                (SCRIPT_MAX_COMMANDS - 1)) {
                                                 edit_line++;
-                                                strcpy(input,
-                                                       print_command(
-                                                           &script[edit]
-                                                                .c[edit_line]));
+                                                print_command(
+                                                    tele_get_script_c(
+                                                        edit, edit_line),
+                                                    input);
                                                 pos = strlen(input);
-                                                for (n = pos; n < 32; n++)
+                                                for (size_t n = pos; n < 32;
+                                                     n++) {
                                                     input[n] = 0;
+                                                }
                                             }
                                         }
                                         else {
-                                            memcpy(&script[edit].c[edit_line],
-                                                   &temp,
-                                                   sizeof(tele_command_t));
-                                            if ((edit_line == script[edit].l) &&
-                                                (script[edit].l <
-                                                 SCRIPT_MAX_COMMANDS))
-                                                script[edit].l++;
+                                            tele_set_script_c(edit, edit_line,
+                                                              &temp);
+                                            if ((edit_line ==
+                                                 tele_get_script_l(edit)) &&
+                                                (tele_get_script_l(edit) <
+                                                 SCRIPT_MAX_COMMANDS)) {
+                                                tele_set_script_l(
+                                                    edit,
+                                                    tele_get_script_l(edit) +
+                                                        1);
```

Comment:
i could make this and similar changes once this is merged, assuming @tehn is okay with the change - i think it would be nice to encapsulate all script related functionality in one file eventually (same for patterns)